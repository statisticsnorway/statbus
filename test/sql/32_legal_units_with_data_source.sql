BEGIN;

\i test/setup.sql

\echo "Setting up Statbus using the web provided examples"

-- A Super User configures statbus.
CALL test.set_user_from_email('test.admin@statbus.org');

\echo "User selected the Activity Category Standard"
INSERT INTO settings(activity_category_standard_id,only_one_setting)
SELECT id, true FROM activity_category_standard WHERE code = 'isic_v4'
ON CONFLICT (only_one_setting)
DO UPDATE SET
   activity_category_standard_id =(SELECT id FROM activity_category_standard WHERE code = 'isic_v4')
   WHERE settings.id = EXCLUDED.id;
;

\echo "User uploads the sample activity categories"
\copy public.activity_category_available_custom(path,name) FROM 'app/public/demo/activity_custom_isic_demo.csv' WITH (FORMAT csv, DELIMITER ',', QUOTE '"', HEADER true);

\echo "User uploads the sample regions"
\copy public.region_upload(path, name) FROM 'app/public/demo/regions_demo.csv' WITH (FORMAT csv, DELIMITER ',', QUOTE '"', HEADER true);

\echo "User uploads the sample legal forms"
\copy public.legal_form_custom_only(code,name) FROM 'app/public/demo/legal_forms_demo.csv' WITH (FORMAT csv, DELIMITER ',', QUOTE '"', HEADER true);

\echo "User uploads the sample sectors"
\copy public.sector_custom_only(path,name,description) FROM 'app/public/demo/sectors_demo.csv' WITH (FORMAT csv, DELIMITER ',', QUOTE '"', HEADER true);

SAVEPOINT before_loading_units;

SELECT
    (SELECT COUNT(DISTINCT id) AS distinct_unit_count FROM public.establishment) AS establishment_count,
    (SELECT COUNT(DISTINCT id) AS distinct_unit_count FROM public.legal_unit) AS legal_unit_count,
    (SELECT COUNT(DISTINCT id) AS distinct_unit_count FROM public.enterprise) AS enterprise_count;

\echo "User uploads the sample legal units"
\copy public.import_legal_unit_current(tax_ident,stat_ident,name,birth_date,physical_region_code,physical_country_iso_2,primary_activity_category_code,legal_form_code,sector_code,employees,turnover,data_source_code) FROM 'app/public/demo/legal_units_demo.csv' WITH (FORMAT csv, DELIMITER ',', QUOTE '"', HEADER true);

SELECT
    (SELECT COUNT(DISTINCT id) AS distinct_unit_count FROM public.establishment) AS establishment_count,
    (SELECT COUNT(DISTINCT id) AS distinct_unit_count FROM public.legal_unit) AS legal_unit_count,
    (SELECT COUNT(DISTINCT id) AS distinct_unit_count FROM public.enterprise) AS enterprise_count;

\echo Run worker processing to run import jobs and generate computed data
CALL worker.process_tasks();
SELECT queue, state, count(*) FROM worker.tasks AS t JOIN worker.command_registry AS c ON t.command = c.command GROUP BY queue,state ORDER BY queue,state;


\echo "Checking statistics"

SELECT unit_type, external_idents, name, data_source_codes, invalid_codes
 FROM statistical_unit
 WHERE valid_after < CURRENT_DATE AND CURRENT_DATE <= valid_to
 ORDER BY unit_type, unit_id, valid_from;

ROLLBACK;
