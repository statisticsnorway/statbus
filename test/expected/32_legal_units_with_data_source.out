BEGIN;
\i test/setup.sql
SELECT * FROM public.statbus_user_create('test.super@statbus.org', 'super_user'::statbus_role_type, 'Super#123!');
         email          |  password  
------------------------+------------
 test.super@statbus.org | Super#123!
(1 row)

SELECT * FROM public.statbus_user_create('test.regular@statbus.org', 'regular_user'::statbus_role_type, 'Regular#123!');
          email           |   password   
--------------------------+--------------
 test.regular@statbus.org | Regular#123!
(1 row)

SELECT * FROM public.statbus_user_create('test.restricted@statbus.org', 'restricted_user'::statbus_role_type, 'Restricted#123!');
            email            |    password     
-----------------------------+-----------------
 test.restricted@statbus.org | Restricted#123!
(1 row)

\if :{?DEBUG}
SET client_min_messages TO debug1;
\else
SET client_min_messages TO NOTICE;
\endif
\echo "Setting up Statbus using the web provided examples"
"Setting up Statbus using the web provided examples"
-- A Super User configures statbus.
CALL test.set_user_from_email('test.super@statbus.org');
\echo "User selected the Activity Category Standard"
"User selected the Activity Category Standard"
INSERT INTO settings(activity_category_standard_id,only_one_setting)
SELECT id, true FROM activity_category_standard WHERE code = 'isic_v4'
ON CONFLICT (only_one_setting)
DO UPDATE SET
   activity_category_standard_id =(SELECT id FROM activity_category_standard WHERE code = 'isic_v4')
   WHERE settings.id = EXCLUDED.id;
;
\echo "User uploads the sample activity categories"
"User uploads the sample activity categories"
\copy public.activity_category_available_custom(path,name) FROM 'app/public/demo/activity_custom_isic_demo.csv' WITH (FORMAT csv, DELIMITER ',', QUOTE '"', HEADER true);
\echo "User uploads the sample regions"
"User uploads the sample regions"
\copy public.region_upload(path, name) FROM 'app/public/demo/regions_demo.csv' WITH (FORMAT csv, DELIMITER ',', QUOTE '"', HEADER true);
\echo "User uploads the sample legal forms"
"User uploads the sample legal forms"
\copy public.legal_form_custom_only(code,name) FROM 'app/public/demo/legal_forms_demo.csv' WITH (FORMAT csv, DELIMITER ',', QUOTE '"', HEADER true);
\echo "User uploads the sample sectors"
"User uploads the sample sectors"
\copy public.sector_custom_only(path,name,description) FROM 'app/public/demo/sectors_demo.csv' WITH (FORMAT csv, DELIMITER ',', QUOTE '"', HEADER true);
SAVEPOINT before_loading_units;
SELECT
    (SELECT COUNT(DISTINCT id) AS distinct_unit_count FROM public.establishment) AS establishment_count,
    (SELECT COUNT(DISTINCT id) AS distinct_unit_count FROM public.legal_unit) AS legal_unit_count,
    (SELECT COUNT(DISTINCT id) AS distinct_unit_count FROM public.enterprise) AS enterprise_count;
 establishment_count | legal_unit_count | enterprise_count 
---------------------+------------------+------------------
                   0 |                0 |                0
(1 row)

\echo "User uploads the sample legal units"
"User uploads the sample legal units"
\copy public.import_legal_unit_current(tax_ident,stat_ident,name,birth_date,physical_region_code,physical_country_iso_2,primary_activity_category_code,legal_form_code,sector_code,employees,turnover,data_source_code) FROM 'app/public/demo/legal_units_demo.csv' WITH (FORMAT csv, DELIMITER ',', QUOTE '"', HEADER true);
SELECT
    (SELECT COUNT(DISTINCT id) AS distinct_unit_count FROM public.establishment) AS establishment_count,
    (SELECT COUNT(DISTINCT id) AS distinct_unit_count FROM public.legal_unit) AS legal_unit_count,
    (SELECT COUNT(DISTINCT id) AS distinct_unit_count FROM public.enterprise) AS enterprise_count;
 establishment_count | legal_unit_count | enterprise_count 
---------------------+------------------+------------------
                   0 |               23 |               23
(1 row)

\echo Run worker processing to generate computed data
Run worker processing to generate computed data
SELECT success, count(*) FROM worker.process_tasks() GROUP BY success;
 success | count 
---------+-------
 t       |     6
(1 row)

\echo "Checking statistics"
"Checking statistics"
SELECT unit_type, external_idents, name, data_source_codes, invalid_codes
 FROM statistical_unit
 WHERE valid_after < CURRENT_DATE AND CURRENT_DATE <= valid_to
 ORDER BY unit_type, unit_id, valid_from;
 unit_type  |                 external_idents                 |            name             | data_source_codes | invalid_codes 
------------+-------------------------------------------------+-----------------------------+-------------------+---------------
 legal_unit | {"tax_ident": "2212760144", "stat_ident": "1"}  | NILE PEARL WATER            | {nlr}             | 
 legal_unit | {"tax_ident": "2812760140", "stat_ident": "2"}  | EQUATOR GLOBE SOLUTIONS     | {nlr}             | 
 legal_unit | {"tax_ident": "2212760084", "stat_ident": "3"}  | EQUATOR CIRCLE CONSTRUCTION | {nlr}             | 
 legal_unit | {"tax_ident": "2612760138", "stat_ident": "4"}  | LIRA FRESH FRUIT EXPORTERS  | {nlr}             | 
 legal_unit | {"tax_ident": "2812760130", "stat_ident": "5"}  | ENTEBBE FUEL ENTERPRISES    | {nlr}             | 
 legal_unit | {"tax_ident": "2312760075", "stat_ident": "6"}  | ZEBRA HAVEN SAFARIS         | {nlr}             | 
 legal_unit | {"tax_ident": "2407020166", "stat_ident": "7"}  | KAMPALA DIGITAL CHARTS      | {nlr}             | 
 legal_unit | {"tax_ident": "2602025168", "stat_ident": "8"}  | MARINE SERVICES             | {nlr}             | 
 legal_unit | {"tax_ident": "2612760118", "stat_ident": "9"}  | NILE TEXTILES               | {nlr}             | 
 legal_unit | {"tax_ident": "2815384160", "stat_ident": "11"} | KAMPALA Cofffe SUPPLIERS    | {nlr}             | 
 legal_unit | {"tax_ident": "2507020167", "stat_ident": "12"} | VICTORIA FRESH FARMS        | {nlr}             | 
 legal_unit | {"tax_ident": "45000", "stat_ident": "13"}      | Statistics Denmark          | {nlr}             | 
 legal_unit | {"tax_ident": "358000", "stat_ident": "14"}     | Statistics Finland          | {nlr}             | 
 legal_unit | {"tax_ident": "46001", "stat_ident": "15"}      | Statistics Sweeden          | {nlr}             | 
 legal_unit | {"tax_ident": "5340521", "stat_ident": "16"}    | Morocco Trade               | {nlr}             | 
 legal_unit | {"tax_ident": "2220794", "stat_ident": "17"}    | Office Picture Corp         | {nlr}             | 
 legal_unit | {"tax_ident": "22501", "stat_ident": "18"}      | Statistics Ethiopia         | {nlr}             | 
 legal_unit | {"tax_ident": "397611", "stat_ident": "19"}     | Erdenes Tavan Tolgoi JSC    | {nlr}             | 
 legal_unit | {"tax_ident": "397622", "stat_ident": "20"}     | Oyu Tolgoi LLC              | {nlr}             | 
 legal_unit | {"tax_ident": "2200019", "stat_ident": "21"}    | Rainbow shopping            | {nlr}             | 
 legal_unit | {"tax_ident": "3300598", "stat_ident": "22"}    | Amman RIAD 13-14            | {nlr}             | 
 legal_unit | {"tax_ident": "4711111", "stat_ident": "230"}   | Statistics Norway           | {nlr}             | 
 legal_unit | {"tax_ident": "3998011", "stat_ident": "231"}   | MANUAL Auto Uzbekistan      | {nlr}             | 
 enterprise | {"tax_ident": "2212760144", "stat_ident": "1"}  | NILE PEARL WATER            | {nlr}             | 
 enterprise | {"tax_ident": "2812760140", "stat_ident": "2"}  | EQUATOR GLOBE SOLUTIONS     | {nlr}             | 
 enterprise | {"tax_ident": "2212760084", "stat_ident": "3"}  | EQUATOR CIRCLE CONSTRUCTION | {nlr}             | 
 enterprise | {"tax_ident": "2612760138", "stat_ident": "4"}  | LIRA FRESH FRUIT EXPORTERS  | {nlr}             | 
 enterprise | {"tax_ident": "2812760130", "stat_ident": "5"}  | ENTEBBE FUEL ENTERPRISES    | {nlr}             | 
 enterprise | {"tax_ident": "2312760075", "stat_ident": "6"}  | ZEBRA HAVEN SAFARIS         | {nlr}             | 
 enterprise | {"tax_ident": "2407020166", "stat_ident": "7"}  | KAMPALA DIGITAL CHARTS      | {nlr}             | 
 enterprise | {"tax_ident": "2602025168", "stat_ident": "8"}  | MARINE SERVICES             | {nlr}             | 
 enterprise | {"tax_ident": "2612760118", "stat_ident": "9"}  | NILE TEXTILES               | {nlr}             | 
 enterprise | {"tax_ident": "2815384160", "stat_ident": "11"} | KAMPALA Cofffe SUPPLIERS    | {nlr}             | 
 enterprise | {"tax_ident": "2507020167", "stat_ident": "12"} | VICTORIA FRESH FARMS        | {nlr}             | 
 enterprise | {"tax_ident": "45000", "stat_ident": "13"}      | Statistics Denmark          | {nlr}             | 
 enterprise | {"tax_ident": "358000", "stat_ident": "14"}     | Statistics Finland          | {nlr}             | 
 enterprise | {"tax_ident": "46001", "stat_ident": "15"}      | Statistics Sweeden          | {nlr}             | 
 enterprise | {"tax_ident": "5340521", "stat_ident": "16"}    | Morocco Trade               | {nlr}             | 
 enterprise | {"tax_ident": "2220794", "stat_ident": "17"}    | Office Picture Corp         | {nlr}             | 
 enterprise | {"tax_ident": "22501", "stat_ident": "18"}      | Statistics Ethiopia         | {nlr}             | 
 enterprise | {"tax_ident": "397611", "stat_ident": "19"}     | Erdenes Tavan Tolgoi JSC    | {nlr}             | 
 enterprise | {"tax_ident": "397622", "stat_ident": "20"}     | Oyu Tolgoi LLC              | {nlr}             | 
 enterprise | {"tax_ident": "2200019", "stat_ident": "21"}    | Rainbow shopping            | {nlr}             | 
 enterprise | {"tax_ident": "3300598", "stat_ident": "22"}    | Amman RIAD 13-14            | {nlr}             | 
 enterprise | {"tax_ident": "4711111", "stat_ident": "230"}   | Statistics Norway           | {nlr}             | 
 enterprise | {"tax_ident": "3998011", "stat_ident": "231"}   | MANUAL Auto Uzbekistan      | {nlr}             | 
(46 rows)

ROLLBACK;
